### DISCUSSION TITLE
bash: ./rv: cannot execute: required file not found

### DISCUSSION THREAD

**Discussion by rubikstriangle** [AcademySoftwareFoundation/OpenRV#90] [General]
It took me way too long this morning to solve the simple issue of this discussion title on Ubuntu 22.10.  The error "cannot execute: required file not found" made me initially jump to the conclusion that my build was missing a file.  Opening up the rv file makes it plainly obvious that that it was tcsh that was missing.  Installing tcsh fixed it.  

Eventually we may have RV installer that installs tcsh if it's not already there, until then, is there a way of making the rv script spit out a warning that tcsh needs to be installed?  Why is rv a tcsh script in the first place?  Don't most linux distributions come with bash as the default?

**Comment by geffrak** [comment]
Perhaps we should convert our `.wrapper` scripts to bash, or get rid of them totally if someone is willing to fix the rpaths. 

I'm surprised that tcsh is not included with Ubuntu.

**Reply by beatreichenbach** [reply]
I rewrote the `rv` wrapper script in bash:
```bash
#!/bin/bash

# Uncomment this if you want to use the pulseaudio/libsndfile package mismatch
# workaround below.
# export RV_USE_PULSEAUDIO_WORKAROUND=1


# Workaround for pulseaudio/libsndfile package mismatch on FC/Cent that causes
# crash when accessing preferences.

if [[ -e /usr/bin/padsp ]] && [[ -n $RV_USE_PULSEAUDIO_WORKAROUND ]]; then
    export OPTIONAL_PADSP="/usr/bin/padsp"
else
    export OPTIONAL_PADSP=""
fi

if [[ -n $QT_PLUGIN_PATH ]]; then
    echo "INFO: warning: QT_PLUGIN_PATH is set, which can cause RV to load the wrong Qt libraries/plugins.  Unsetting..."
    unset QT_PLUGIN_PATH
fi

# Uncomment this if you're on an older linux distro and RV is hanging in
# the Audio preferences or on startup with audio sources
# export PA_ALSA_EXCLUDE_DMIX_DEFAULT=1


# For unknown reasons, LANG causes problems when set to
# interesting values (like fr_FR.UTF-8).
unset LANG

if [[ -z $RV_HOME ]]; then
    current_dir=$(dirname "$(readlink -f "$0")" )
    root_dir=$(dirname "$current_dir" )
    export RV_HOME="$root_dir"
fi

platform=i386
rvbin="$RV_HOME/bin/rv.bin.$platform"

if [[ ! -e "$rvbin" ]]; then
    rvbin="$RV_HOME/bin/rv.bin"
fi

unset BUILD_ROOT
export PATH="$RV_HOME/bin:${PATH}"

if [[ -n $LD_LIBRARY_PATH ]]; then
     export LD_LIBRARY_PATH="$RV_HOME/lib:$LD_LIBRARY_PATH"
else
     export LD_LIBRARY_PATH="$RV_HOME/lib"
fi
echo "$LD_LIBRARY_PATH"

# Unless the RV_USE_SYSTEM_OPENSSL environment variable is set, use the
# OpenSSL provided with RV when the required OpenSSL version cannot be found.
#
if [[ -z $RV_USE_SYSTEM_OPENSSL ]]; then
     required_openssl_found=$( openssl version | grep 1.1.1 )
     if [[ -z $required_openssl_found ]]; then
        export LD_LIBRARY_PATH="$RV_HOME/lib/OpenSSL:$LD_LIBRARY_PATH"
     fi
fi

# exec RV
if [[ "$1" == "-d" || "$1" == "--debug" ]]; then
  shift  # skip '-d / --debug'
    gdb "$OPTIONAL_PADSP$rvbin" "$@"
else
    exec "$OPTIONAL_PADSP$rvbin" "$@"
fi
```

**Reply by geffrak** [reply]
This is great!

Can you make it a pull request? Including the other  wrapper scripts.

**Reply by beatreichenbach** [reply]
Yeah, I first want to get it all working (https://github.com/AcademySoftwareFoundation/OpenRV/discussions/382), and then I'll do that with additional info for my build process.

**Comment by jorxster** [comment]
@rubikstriangle , have you managed to build on Ubuntu 22?

**Reply by rubikstriangle** [reply]
No, I ended up spinning up centos7 in a virtualbox and building there.  Works fine in Ubuntu.
